app/services/orchestration/task.rb

```
@@ -1,12 +1,17 @@
 module Orchestration
   class Task
-    attr_reader :name, :status, :priority, :action, :timestamp
+    attr_reader :id, :name, :status, :priority, :action, :timestamp, :created
 
     def initialize(opts)
       @name      = opts[:name]
+      @id        = opts[:id].try(:to_s) || @name
       @status    = opts[:status]
       @priority  = opts[:priority] || 0
       @action    = opts[:action]
+      @created   = opts[:created] || Time.now.to_f
+      raise("action must be present for task '#{name}'") if action.nil?
+      # this kills tests, need to update other tests before
```

what tests you had in mind? and why action is now required? I mean is it refactoring so we fail faster if action is missing?

---

The comment "this kills" was meant for line 14, deprecated exceptions actually make tests to fail. Re: action - since all attributes are read only, it makes no sense to create such a task?